docs/topics/multiplatform/multiplatform-compatibility-guide.md

```diff
@@ -926,48 +926,49 @@ and prepare this functionality for the upcoming Kotlin 2.0 release. From now on:
   Now, you can only use a special forward declaration package for that: `import cnames.structs.cstructName`.
   The same is true for `objcnames`.
 
-*  Consider two objcinterop libraries, one that uses `objcnames.protocols.ForwardDeclaredProtocolProtocol` and the other that has an actual definition:
-
-    ```ObjC
-    // First objcinterop library
-    #import <Foundation/Foundation.h>
-    
-    @protocol ForwardDeclaredProtocol;
-    
-    NSString* consumeProtocol(id<ForwardDeclaredProtocol> s) {
-        return [NSString stringWithUTF8String:"Protocol"];
-    }
-    ```
-    
-    ```ObjC
-    // Second objcinterop library
-    // Header:
-    #import <Foundation/Foundation.h>
-    @protocol ForwardDeclaredProtocol
-    @end
-    // Implementation:
-    @implementation ForwardDeclaredProtocolImpl : NSObject
-    @end;
-    
-    id<ForwardDeclaredProtocol> produceProtocol() {
-        return [ForwardDeclaredProtocolImpl new];
-    }
-    ```
-    
-    Previously, it was possible to transfer objects between them seamlessly. Now, an explicit `as`  cast is required
-    for the forward declaration:
-    
-    ```kotlin
-    // Kotlin code:
-    fun test() {
-        consumeProtocol(produceProtocol() as objcnames.protocols.ForwardDeclaredProtocolProtocol)
-    }
-    ```
-    
-    > The casting to `objcnames.protocols.ForwardDeclaredProtocolProtocol` is only allowed from the corresponding real class.
-    > Otherwise, you'll get an error.
-    >
-    {style="note"}
+* Consider two objcinterop libraries, one that uses `objcnames.protocols.ForwardDeclaredProtocolProtocol` and the other
+  that has an actual definition:
+
+  ```ObjC
+  // First objcinterop library
+  #import <Foundation/Foundation.h>
+  
+  @protocol ForwardDeclaredProtocol;
+  
+  NSString* consumeProtocol(id<ForwardDeclaredProtocol> s) {
+      return [NSString stringWithUTF8String:"Protocol"];
+  }
+  ```
+
+  ```ObjC
+  // Second objcinterop library
+  // Header:
+  #import <Foundation/Foundation.h>
+  @protocol ForwardDeclaredProtocol
+  @end
+  // Implementation:
+  @interface ForwardDeclaredProtocolImpl : NSObject <ForwardDeclaredProtocol>
+  @end
+
+  id<ForwardDeclaredProtocol> produceProtocol() {
+      return [ForwardDeclaredProtocolImpl new];
+  }
+  ```
+
+  Previously, it was possible to transfer objects between them seamlessly. Now, an explicit `as` cast is required
+  for the forward declaration:
+
+  ```kotlin
+  // Kotlin code:
+  fun test() {
+      consumeProtocol(produceProtocol() as objcnames.protocols.ForwardDeclaredProtocol)
```

Comment:
Oh, actually `ForwardDeclaredProtocolProtocol` was the correct name. Sorry I [misled](https://github.com/JetBrains/kotlin-web-site/pull/4477#discussion_r1790357578) you before.

Alternatively, we can declare it as `@protocol ForwardDeclared`, and the name in Kotlin will be `ForwardDeclaredProtocol`.

Reply:
Similar problem in the cast in the other document, `native-objc-interop.md`.

Reply:
Thanks! Let's revert to the previous naming